lib/Pod/Markdown.pm

```diff
@@ -1224,16 +1224,22 @@ and L</markdown_fragment_format> is used (which can be customized).
 sub format_perldoc_url {
   my ($self, $name, $section) = @_;
 
-  my $url_prefix = defined($name)
+  my $url_prefix = $self->perldoc_url_prefix;
+  if (
+    defined($name)
     && $self->is_local_module($name)
-    && $self->local_module_url_prefix
-    || $self->perldoc_url_prefix;
+    && defined($self->local_module_url_prefix)
+  ) {
+    $url_prefix = $self->local_module_url_prefix;
+  }
 
   my $url = '';
 
   # If the link is to another module (external link).
   if ($name) {
     $url = $url_prefix . $name;
+    # If there's no prefix don't let the name to be interpreted as an ipv6 address.
+    $url =~ s/::/%3a%3a/g if $url_prefix eq '';
```

Comment:
this isn't great, but i wasn't sure if i should change the behavior of everything else... and the existing entity encode function wasn't super open to extension (read: i'm lazy).

Reply:
Would it be too extreme to pass every module name, whether it's local or not, through URI::Escape? That would cover other surprising characters too.  Just escaping :: is *probably* enough though...

Reply:
probably not... aside from it not looking as pretty in the source, and being a new dependency, it seems like it make the most sense.

Reply:
Thanks for mentioning URI::Escape... i hadn't yet differentiated between html and uri escaping... but now i don't have to go looking for which module i should use :-)

Reply:
Thanks @karenetheridge! i updated the README to test it out and that seems like a good change.

Reply:
++

Reply:
Thanks for the 👀 !